File: Review/Python/review-8.py
class Student:
    def __init__(self, name, korean, english, math):
        self.name = name
        self.korean = korean
        self.english = english
        self.math = math
        self.avg = (korean + english + math) / 3.0

def main():
    N = int(input())
    
    students = []
    for _ in range(N):
        name, korean, english, math = input().split()
        korean, english, math = int(korean), int(english), int(math)
        students.append(Student(name, korean, english, math))
    
    for student in students:
        print(f"{student.name} {student.avg:.1f}", end=" ")

        if student.korean >= 90 or student.english >= 90 or student.math >= 90:
            print("GREAT", end=" ")
        
        if student.korean < 70 or student.english < 70 or student.math < 70:
            print("BAD", end=" ")
        
        print()

if __name__ == '__main__':
    main()

File: Review/C/review-8.c
#pragma warning(disable:4996)
#include <stdio.h>
#include <stdlib.h>

typedef struct Student {
    char name[8];
    int korean, english, math;
    double avg;
}Student;

int main() {

    int N;
    scanf("%d", &N);

    Student *students = (Student *)malloc(sizeof(Student) * N);
    if(students == NULL) {
        printf("students 동적 할당 실패\n");
        return 0;
    }

    for(int i = 0; i < N; i++) {
        scanf("%s %d %d %d", students[i].name, &students[i].korean, &students[i].english, &students[i].math);
        students[i].avg = (students[i].korean + students[i].english + students[i].math) / 3.0;
    }

    for(int i = 0; i < N; i++) {
        printf("%s %.1lf ", students[i].name, students[i].avg);
        if (students[i].korean >= 90 || students[i].english >= 90 || students[i].math >= 90) 
			printf("GREAT ");
		
		if (students[i].korean < 70 || students[i].english < 70 || students[i].math < 70)
			printf("BAD");
        printf("\n");
    }

    free(students);

    return 0;
}